### PR TITLE
AF samples feed

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22358

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [X] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py [hunk 1]
```diff
@@ -182,15 +367,22 @@ def build_iterator(self, limit=None, offset=None):
                     }
                 })
 
-        return parsed_indicators
+        if "Sample Feed" in self.indicator_feeds:
+            parsed_indicators.extend(self.sample_http_request())
+
+        if limit:
+            parsed_indicators = parsed_indicators[int(offset): int(offset) + int(limit)]
```
Comment (ronykoz) [inline Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: why do we need this limit?

**Comment by ronykoz** [comment]
Ping me when it's review ready

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 1 above)
Comment (teizenman) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: its there for the get_indicators_command, not for the feed itself

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.yml [hunk 2]
```diff
@@ -25,6 +26,20 @@ configuration:
...
+- additionalinfo: Only necessary in case a Sample Feed is fetched.
+  display: Sample Feed Scope Type
+  name: scope_type
+  options:
+  - public
+  - private
+  - global
+  required: false
+  type: 15
+- additionalinfo: JSON styled AutoFocus query. Only necessary in case a Sample Feed is fetched.
```
Comment (ronykoz) [inline Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.yml]: Can we provide a link or an example?

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 1 above)
Comment (ronykoz) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: lets add a comment for this

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py [hunk 3]
```diff
@@ -86,13 +120,138 @@ def http_request(self, feed_type) -> list:
...
+            method="POST",
+            headers={'Content-Type': "application/json"},
+            url=SAMPLE_FEED_REQUEST_BASE_URL,
+            verify=self.verify,
+            json=request_body
+        )
+        initiate_sample_res.raise_for_status()
+
+        af_cookie = initiate_sample_res.json()['af_cookie']
+        time.sleep(20)
```
Comment (ronykoz) [inline Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: why?

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py [hunk 4]
```diff
@@ -86,13 +120,138 @@ def http_request(self, feed_type) -> list:
...
+        if create_date is not None:
+            create_date = datetime.strptime(create_date, '%Y-%m-%dT%H:%M:%S')
+            raw_json_data['autofocus_create_date'] = datetime_to_epoch(create_date)
+
+        sha256 = single_sample_data.get('sha256', '...')
+        sha1 = single_sample_data.get('sha1', None)
+        md5 = single_sample_data.get('md5')
+
+        if '...' in sha256:
+            return []
```
Comment (ronykoz) [inline Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: add an explanation note

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py [hunk 5]
```diff
@@ -86,13 +120,138 @@ def http_request(self, feed_type) -> list:
...
+                'type': FeedIndicatorType.File,
+                'rawJSON': raw_json_data
+            }
+            for file_indicator in [sha256, sha1, md5] if file_indicator
+        ]
+
+        for artifact in artifacts:
+            indicator_from_artifact = Client.create_indicator_from_artifact(raw_json_data, artifact)
+            if indicator_from_artifact:
+                indicators.append(indicator_from_artifact)
```
Comment (ronykoz) [inline Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: didn't we say each artifact is a list? so it should return a list of indicators, doesn't it?

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.yml (hunk 2 above)
Comment (teizenman) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.yml]: example added to description

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 3 above)
Comment (teizenman) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: The api needs some time to process the request, thats a random number that seemed to be working at all times, lets talk about it tomorrow morning

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 4 above)
Comment (teizenman) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: Lets talk about that too

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 5 above)
Comment (teizenman) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: Each artifact is an indicator.
The source is the big file which holds all the artifacts.

File: Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py (hunk 4 above)
Comment (ronykoz) [reply Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py]: did we check that with Luigi?

**Comment by teizenman** [comment]
@kirbles19 Can you please go over the following files:

- https://github.com/demisto/content/blob/af-private/Packs/FeedAutofocus/Integrations/FeedAutofocus/CHANGELOG.md
- params `sample_query`, `scope_type` at https://github.com/demisto/content/blob/af-private/Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.yml
- `Samples feed` part at https://github.com/demisto/content/blob/af-private/Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus_description.md

**Comment by kirbles19** [comment]
@teizenman Done.

**Comment by teizenman** [comment]
@kirbles19 can you go over the changes in the README.md: https://github.com/demisto/content/blob/af-private/Packs/FeedAutofocus/Integrations/FeedAutofocus/README.md

The rest was not touched since your approval

**Comment by kirbles19** [comment]
@teizenman Done, no changes, looked good.

**Pull request merged**